### PR TITLE
fix: add proper cors validation on api and allow specific endpoints

### PR DESCRIPTION
**Pull request by Ameer-officials**
/claim #4155
## Related issues

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Enhancement (non-breaking small changes to existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

**Comment by bigint** [comment]
How can i test this?

**Comment by bigint** [comment]
Fix the PR title

File: apps/api/src/utils/express/corsConfig.ts [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
-      origin.includes('heyxyz.vercel.app') ||
-      origin.startsWith('http://localhost')
+      /^https:\/\/(\w+\.)?hey\.xyz$/.test(origin) || // https://hey.xyz and https://*.hey.xyz
+      /^https:\/\/\w+-heyxyz\.vercel\.app$/.test(origin) || // https://*-heyxyz.vercel.app
+      origin === "http://localhost:4783"
     ) {
       callback(null, true);
     } else {
-      callback(new Error('Not allowed by CORS'));
+      callback(new Error("Not allowed by CORS"));
```
Comment (bigint) [inline apps/api/src/utils/express/corsConfig.ts:13]: Can we also return with proper cors error or not allowed error instead of 500?

**Comment by Ameer-officials** [comment]
Greetings,
Ok im working on it,
i was outside

**Pull request closed without merging**